### PR TITLE
GLAM rewrite JS UDFs in SQL

### PR DESCRIPTION
**Pull request by edugfilho**
Fixes https://github.com/mozilla/glam/issues/2840
@BenWu's suggestion in the issue above was to set up small tests where we'd replace the JS UDFs with something that does very little and see at best the potential amount of performance to gain. 

Despite agreeing with the approach I thought it'd be a good idea to make the conversions at once, especially for `functional_buckets`, which will probably be more used once [Glean stops sending 0-count buckets](https://bugzilla.mozilla.org/show_bug.cgi?id=1898336). 
And since there are some tests in the UDFs, I decided to take Ben's [samples](https://github.com/mozilla/bigquery-etl/pull/5689) and converted the other GLAM UDFs. In fact there's only one UDF that hasn't been converted to SQL, because it's not trivial [(here in case someone wants to take a shot at it)](https://github.com/mozilla/bigquery-etl/blob/main/sql/mozfun/glam/histogram_generate_exponential_buckets/udf.sql)

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3947)

### CODE REVIEW

File: sql/mozfun/glam/percentile/udf.sql [hunk 1]
```diff
@@ -1,37 +1,41 @@
...
+          pct >= 0
+          AND pct <= 100,
+          TRUE,
+          ERROR('percentile must be a value between 0 and 100')
+        ) pct_ok
+    ),
+    keyed_cum_sum AS (
+      SELECT
+        key,
+        SUM(value) OVER (ORDER BY key) / SUM(value) OVER () AS cum_sum
```
Comment (BenWu) [inline sql/mozfun/glam/percentile/udf.sql:21]: I just learned it defaults to `RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW` if it's ordered

**Review by BenWu — COMMENTED** [review]
Good to know that the other ones aren't too hard to convert. Looks like the sql tests are failing at least partially due to the ones I wrote.  
Have you tested these on the real data to see if the output looks ok and there are improvements?  Getting them wrong and needing a backfill could be expensive and the unit tests aren't the most comprehensive

**Comment by edugfilho** [comment]
> Good to know that the other ones aren't too hard to convert. Looks like the sql tests are failing at least partially due to the ones I wrote. Have you tested these on the real data to see if the output looks ok and there are improvements? Getting them wrong and needing a backfill could be expensive and the unit tests aren't the most comprehensive

I haven't done extensive test. I'll test it, fix the code and add more tests.

**Comment by edugfilho** [comment]
The tests have been fixed and I tested the code with a number of instances from real data by using expressions such as `assert.equal(<original_udf_in_js>(input), <new_udf_in_sql>(input))`

**Review by BenWu — APPROVED** [review]
LGTM but it might not be ideal to merge this on a Friday unless you can make sure it works in prod today

**Pull request merged**